tests/test_motiondetector.py

```diff
@@ -8,29 +8,80 @@
 from viam.services.vision import CaptureAllResult, Classification, Detection
 from typing import List, Mapping, Any
 
+from parameterized import parameterized
 import pytest
 import cv2
 import numpy as np
 
+
 def make_component_config(dictionary: Mapping[str, Any]) -> ComponentConfig:
         struct = Struct()
         struct.update(dictionary=dictionary)
         return ComponentConfig(attributes=struct)
 
 
+def getMD():
+    md = MotionDetector("test")
+    md.sensitivity = 0.9
+    md.min_box_size    = 1000
+    md.min_box_percent = None
+    md.max_box_size    = None
+    md.max_box_percent = None
+    md.cam_name = "test"
+    md.camera = FakeCamera("test")
+    return md
+
+
+class TestConfigValidation:
+    @parameterized.expand((
+        ("all defaults",                     {}),
+        ("min in pixels",                    {"min_box_size": 3}),
+        ("min in percentage",                {"min_box_percent": 0.1}),
+        ("max in pixels",                    {"max_box_size": 300}),
+        ("max in percentage",                {"max_box_percent": 0.9}),
+        ("min and max in pixels",            {"min_box_size": 3, "max_box_size": 300}),
+        ("min in pixels, max in percentage", {"min_box_size": 3, "max_box_percent": 0.9}),
+        ("min and max in percentage",        {"min_box_percent": 0.1, "max_box_percent": 0.9}),
+        ("min in percentage, max in pixels", {"min_box_percent": 0.1, "max_box_size": 300}),
+    ))
+    def test_valid(self, unused_test_name, extra_config_values):
+        md = getMD()
+        raw_config = {"cam_name": "test"}
+        raw_config.update(extra_config_values)
+        config = make_component_config(raw_config)
+        response = md.validate_config(config=config)
+        assert response == ["test"]
+
+
+    @parameterized.expand((
+        ({"min_box_size": -1},      "Minimum bounding box size should be a non-negative integer"),
+        ({"min_box_percent": -0.1}, "Minimum bounding box percent should be between 0.0 and 1.0"),
+        ({"min_box_percent": 1.1},  "Minimum bounding box percent should be between 0.0 and 1.0"),
+        ({"max_box_size": -1},      "Maximum bounding box size should be a non-negative integer"),
+        ({"max_box_percent": -0.1}, "Maximum bounding box percent should be between 0.0 and 1.0"),
+        ({"max_box_percent": 1.1},  "Maximum bounding box percent should be between 0.0 and 1.0"),
+        ({"min_box_size": 3, "min_box_percent": 0.1},
+            "Cannot specify the minimum box in both pixels and percentages"),
+        ({"max_box_size": 300, "max_box_percent": 0.9},
+            "Cannot specify the maximum box in both pixels and percentages"),
+    ))
+    def test_invalid(self, extra_config_values, error_message):
```

Comment:
Swap the arguments around, so that the generated test names are easier to read (names are based off the string representation of the first argument)